src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -43,6 +43,20 @@
 
 logger = getLogger(__name__)
 
+# We allow some common aliases to be used
+OPTION_ALIASES = {
+    "HEADER": ("SKIP_HEADER", lambda val: 1 if val else 0),
```

Comment:
Just FYI, we are introducing another option called `PARSE_HEADER` which will parse the first record and persist the column names. But it's only working for limited number of features rn

Reply:
https://docs.snowflake.com/en/sql-reference/functions/infer_schema#usage-notes

Reply:
In spark header is pretty basic, it is just to skip first line